### PR TITLE
fix mill build

### PR DESCRIPTION
**Pull request by sequencer**
**Related issue**: #1291 #1035

**Type of change**: bug report

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**
Bug fix for mill build system, original mill has been deprecated since the update of mill.
This implementation use native mill API as much as possible, which can keep better compatibility to which.

### CODE REVIEW

**Comment by sequencer** [comment]
As for accessing by other `build.sc`, for example there is a project structure in: 
```
|proj
|-chisel3
|-firrtl
```
we can give a `build.sc` under proj,
```
import $file.chisel3.build
import $file.firrtl.build

object myfirrtl extends firrtl.build.firrtlCrossModule("2.12.10") {
  override def millSourcePath = super.millSourcePath / 'firrtl
}

object mychisel3 extends chisel3.build.chisel3CrossModule("2.12.10") {
  override def millSourcePath = super.millSourcePath / 'chisel3
  def firrtlModule: Option[PublishModule] = Some(myfirrtl)
}

object foo extends ScalaModule {
  def moduleDeps = Seq(mychisel3)
}
```
to make `mychisel3` depend on `myfirrtl`. Then `foo` can depend on the `mychisel3`.

File: build.sc [hunk 1]
```diff
@@ -1,249 +1,124 @@
...
+
+  override def compileIvyDeps = Agg(macroParadise)
+
+  override def scalacPluginIvyDeps = Agg(macroParadise)
 
   def pomSettings = PomSettings(
     description = artifactName(),
     organization = "edu.berkeley.cs",
-    url = "https://chisel.eecs.berkeley.edu",
+    url = "https://github.com/freechipsproject/chisel3",
```
Comment (edwardcwang) [inline build.sc]:
```suggestion
    url = "https://www.chisel-lang.org/",
```
?

File: build.sc [hunk 2]
```diff
@@ -1,249 +1,134 @@
-import ammonite.ops._
-import ammonite.ops.ImplicitWd._
-import coursier.MavenRepository
 import mill._
```
Comment (edwardcwang) [inline build.sc]: Maybe add a line above this saying what mill version this has been tested with?

**Comment by sequencer** [comment]
Any update for this PR?

**Pull request merged**